acra-censor/handlers/querycapture_handler.go

```diff
@@ -192,25 +148,112 @@ func (handler *QueryCaptureHandler) GetForbiddenQueries() []string {
 	}
 	return forbiddenQueries
 }
+
 func (handler *QueryCaptureHandler) SetSerializationTimeout(timeout time.Duration) {
 	handler.serializationTimeout = timeout
 }
+
 func (handler *QueryCaptureHandler) GetSerializationTimeout() time.Duration {
 	return handler.serializationTimeout
 }
 
-func (handler *QueryCaptureHandler) Serialize() error {
-	jsonFile, err := json.Marshal(handler.Queries)
+func (handler *QueryCaptureHandler) DumpAllQueriesToFile() error {
+	// open or create file, NO APPEND
+	f, err := os.OpenFile(handler.filePath, os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.WithError(ErrCantOpenFileError).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorIOError)
+		return err
+	}
+
+	// write all queries
+	return AppendQueries(handler.Queries, f)
+}
+
+func (handler *QueryCaptureHandler) DumpBufferedQueriesToFile(openedFile *os.File) error {
+	// write buffered queries
+	buf := handler.BufferedQueries
+
+	// nothing to dump
+	if len(buf) == 0 {
+		return nil
+	}
+
+
+	// clean buffered queries
+	handler.BufferedQueries = nil
+	err := AppendQueries(buf, openedFile)
```

Comment:
we can just `return AppendQueries(buf, openedFile)` 
or `return err` below without `if`